applications/tests/test_applications_new_schema_ws_queries.py

```diff
@@ -0,0 +1,407 @@
+import random
+
+import pytest
+from dateutil.parser import isoparse
+from freezegun import freeze_time
+from graphql_relay.node.node import to_global_id
+
+from applications.enums import ApplicationStatus
+from applications.new_schema.types import WinterStorageApplicationNode
+from applications.tests.factories import (
+    WinterAreaChoiceFactory,
+    WinterStorageApplicationFactory,
+)
+from berth_reservations.tests.utils import assert_in_errors
+from leases.tests.factories import WinterStorageLeaseFactory
+
+WS_APPLICATIONS_WITH_NO_CUSTOMER_FILTER_QUERY = """
+query APPLICATIONS {
+    winterStorageApplications(noCustomer: %s) {
+        edges {
+            node {
+                id
+                customer {
+                    id
+                }
+            }
+        }
+    }
+}
+"""
+
+
+@pytest.mark.parametrize(
+    "api_client",
+    ["berth_services", "berth_handler", "berth_supervisor"],
+    indirect=True,
+)
+def test_winter_storage_applications_no_customer_filter_true(
+    winter_storage_application, api_client
+):
+    winter_storage_application.customer = None
+    winter_storage_application.save()
+
+    query = WS_APPLICATIONS_WITH_NO_CUSTOMER_FILTER_QUERY % "true"
+
+    executed = api_client.execute(query)
+
+    assert executed["data"] == {
+        "winterStorageApplications": {
+            "edges": [
+                {
+                    "node": {
+                        "id": to_global_id(
+                            WinterStorageApplicationNode._meta.name,
+                            winter_storage_application.id,
+                        ),
+                        "customer": None,
+                    }
+                }
+            ]
+        }
+    }
+
+
+@pytest.mark.parametrize(
+    "api_client",
+    ["berth_services", "berth_handler", "berth_supervisor"],
+    indirect=True,
+)
+def test_winter_storage_applications_no_customer_filter_false(
+    winter_storage_application, api_client
+):
+    winter_storage_application.customer = None
+    winter_storage_application.save()
+
+    query = WS_APPLICATIONS_WITH_NO_CUSTOMER_FILTER_QUERY % "false"
+
+    executed = api_client.execute(query)
+
+    assert executed["data"] == {"winterStorageApplications": {"edges": []}}
+
+
+WINTER_STORAGE_APPLICATIONS_WITH_STATUSES_FILTER_QUERY = """
+query APPLICATIONS {
+    winterStorageApplications(statuses: [%s]) {
+        edges {
+            node {
+                id
+                status
+            }
+        }
+    }
+}
+"""
+
+
+def create_handled_application(winter_storage_application) -> None:
+    winter_storage_application.lease = WinterStorageLeaseFactory()
+    winter_storage_application.status = ApplicationStatus.HANDLED
+    winter_storage_application.save()
```

Comment:
might make sense to just create a fixture out of it, it could look like this if you put it after your `winter_storage_application` fixture:
```python
@pytest.fixture
def handled_ws_application(winter_storage_application):
    winter_storage_application.status = ApplicationStatus.HANDLED
    winter_storage_application.save()
    return winter_storage_application
```